contracts/src/v0.8/vrf/VRFCoordinatorV2Plus.sol

```diff
@@ -0,0 +1,611 @@
+  uint256 private constant GAS_FOR_CALL_EXACT_CHECK = 5_000;
+  error InvalidRequestConfirmations(uint16 have, uint16 min, uint16 max);
+  error GasLimitTooBig(uint32 have, uint32 want);
+  error NumWordsTooBig(uint32 have, uint32 want);
+  error ProvingKeyAlreadyRegistered(bytes32 keyHash);
+  error NoSuchProvingKey(bytes32 keyHash);
+  error InvalidLinkWeiPrice(int256 linkWei);
+  error InsufficientGasForConsumer(uint256 have, uint256 want);
+  error NoCorrespondingRequest();
+  error IncorrectCommitment();
+  error BlockhashNotInStore(uint256 blockNum);
+  error PaymentTooLarge();
+  struct RequestCommitment {
+    uint64 blockNum;
+    uint64 subId;
+    uint32 callbackGasLimit;
+    uint32 numWords;
+    address sender;
+    bool nativePayment;
+  }
+  mapping(bytes32 => address) /* keyHash */ /* oracle */ public s_provingKeys;
+  bytes32[] public s_provingKeyHashes;
+  mapping(uint256 => bytes32) /* requestID */ /* commitment */ public s_requestCommitments;
+  event ProvingKeyRegistered(bytes32 keyHash, address indexed oracle);
+  event ProvingKeyDeregistered(bytes32 keyHash, address indexed oracle);
+  event RandomWordsRequested(
+    bytes32 indexed keyHash,
+    uint256 requestId,
+    uint256 preSeed,
+    uint64 indexed subId,
+    uint16 minimumRequestConfirmations,
+    uint32 callbackGasLimit,
+    uint32 numWords,
+    bool nativePayment,
+    address indexed sender
+  );
+  event RandomWordsFulfilled(
+    uint256 indexed requestId,
+    uint256 outputSeed,
+    uint96 payment,
+    bool nativePayment,
+    bool success
+  );
+
+  struct Config {
+    uint16 minimumRequestConfirmations;
+    uint32 maxGasLimit;
+    // stalenessSeconds is how long before we consider the feed price to be stale
+    // and fallback to fallbackWeiPerUnitLink.
+    uint32 stalenessSeconds;
+    // Gas to cover oracle payment after we calculate the payment.
+    // We make it configurable in case those operations are repriced.
+    // The recommended number is below, though it may vary slightly
+    // if certain chains do not implement certain EIP's.
+    // 21000 + // base cost of the transaction
+    // 100 + 5000 + // warm subscription balance read and update. See https://eips.ethereum.org/EIPS/eip-2929
+    // 2*2100 + 5000 - // cold read oracle address and oracle balance and first time oracle balance update, note first time will be 20k, but 5k subsequently
+    // 4800 + // request delete refund (refunds happen after execution), note pre-london fork was 15k. See https://eips.ethereum.org/EIPS/eip-3529
+    // 6685 + // Positive static costs of argument encoding etc. note that it varies by +/- x*12 for every x bytes of non-zero data in the proof.
+    // 21000  // cost of a cold storage write to update the request payments mapping.
+    // Total: 58,815 gas.
+    uint32 gasAfterPaymentCalculation;
+  }
+  int256 public s_fallbackWeiPerUnitLink;
+  Config public s_config;
+  FeeConfig public s_feeConfig;
+  struct FeeConfig {
```

Comment:
So no request tiers now.

Reply:
I've removed them but we can stick them back in. AFAIK we never used them for V2.

Reply:
Yeah no need to have useless fields, just confirming intentional.